Makefile

```
@@ -19,40 +19,49 @@ dev: ## Setup a development environment
 
 .PHONY: build
 build: ## Build the project, including non installable libraries and executables
-	dune build --root .
+	dune build @all
 
 .PHONY: install
 install: all ## Install the packages on the system
-	dune install --root .
+	dune install
 
 .PHONY: lock
 lock: ## Generate the lock files
 	opam lock -y .
 
 .PHONY: test
-test: ## Run the unit tests
-	dune build --root . @lsp-fiber/runtest @fiber-unix/runtest @jsonrpc-fiber/runtest
+test-ocaml: ## Run the unit tests
+	dune build @lsp-fiber/runtest @fiber-unix/runtest @jsonrpc-fiber/runtest
 
-.PHONY: test-template
+.PHONY: promote
+	dune promote
+
+.PHONY: check
+	dune build @check
+
+.PHONY: test-e2e
 test-e2e: $(TEST_E2E_DIR)/node_modules all ## Run the template integration tests
-	cd $(TEST_E2E_DIR) && dune exec -- yarn test
+	dune build @install && cd $(TEST_E2E_DIR) && dune exec -- yarn test
```

The `all` target (on which this target depends) already runs `dune build @install` so I don't think it's needed here

```suggestion
	cd $(TEST_E2E_DIR) && dune exec -- yarn test
```

---

Ah yes. But it's weird that `all` actually builds `@install`. I changed `all` to build `@all` and this target should depend on `@install`. Running the tests should only require the installed targets.